src/hotspot/share/opto/machnode.cpp

```diff
@@ -806,6 +806,18 @@ void MachCallDynamicJavaNode::dump_spec(outputStream *st) const {
 }
 #endif
 //=============================================================================
+const RegMask &MachCallBlackholeNode::in_RegMask(uint idx) const {
+  // Fake the incoming arguments mask for blackholes: accept all registers and all stack slots.
+  // This must avoid moving the arguments for the call that never happens.
+  return RegMask::All;
+}
+#ifndef PRODUCT
+void MachCallBlackholeNode::dump_spec(outputStream *st) const {
+  st->print("Blackhole ");
+  MachCallNode::dump_spec(st);
```

Comment:
How does the output look like? It would be nice to have the info what are the values being consumed and where they are located.

Reply:
I am not sure what takes this path. I modeled this after other `MachCall*Node`-s, by delegating to `MachCallNode`, plus the type message. I thought that would be enough to dump the superclass-call-specific spec.

Reply:
`Node::dump()` eventually calls it. Also, IGV collects and shows it (thought it's trimmed right now). 
There's no flag to dump Mach IR except `-XX:+PrintOptoAssembly`, but it shows different info.
Thinking more about it, I don't see much benefit in improving `MachCallBlackholeNode::dump()` since it lacks some important information.

Considering blackholes don't obey standard calling conventions, it's hard to say where the arguments are located in generated code.   It would be nice to enrich `-XX:+PrintOptoAssembly` with information about consumed values. `MachNode::format` has access to such information through `PhaseRegAlloc* ra` argument.